### PR TITLE
Added Biome Tags

### PR DESCRIPTION
**Pull request by GirafiStudios**
Based on the custom tag type system by @pupnewfster, added in https://github.com/MinecraftForge/MinecraftForge/pull/7289

This pull request aims to be a replacement for the BiomeDictionary, in the form of tags.
Simply just adds the same tags, as the BiomeDictionary have been using previously. Also adds a Biome Tag data generator, to easily add or modify biomes added to tags.

Biomes has the exact same tags, as they did in the BiomeDictionary. Although I did make all End biomes, have the same tags as the "The End"- biome had already assigned.
I also added appropriate tags to all new nether biomes.

There is still a few issues that needs to be sorted out, in regards to issues with biomes having both a ForgeRegistry & the dynamic registry. A fix for that is currently being worked on in https://github.com/MinecraftForge/MinecraftForge/pull/7316. Keeping this as a draft in the meantime.

### CODE REVIEW

File: src/generated/resources/data/forge/tags/biomes/overworld.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    "minecraft:dark_forest_hills",
+    "minecraft:snowy_taiga_mountains",
+    "minecraft:giant_spruce_taiga",
+    "minecraft:giant_spruce_taiga_hills",
+    "minecraft:modified_gravelly_mountains",
+    "minecraft:shattered_savanna",
+    "minecraft:shattered_savanna_plateau",
+    "minecraft:eroded_badlands",
+    "minecraft:modified_wooded_badlands_plateau",
+    "minecraft:modified_badlands_plateau"
```
Comment (bageldotjpg) [inline src/generated/resources/data/forge/tags/biomes/overworld.json]: Missing `bamboo_jungle` and `bamboo_jungle_hills`

File: src/generated/resources/data/forge/tags/biomes/wet.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "replace": false,
+  "values": [
+    "minecraft:swamp",
+    "minecraft:jungle",
+    "minecraft:jungle_hills",
+    "minecraft:jungle_edge",
+    "minecraft:swamp_hills",
+    "minecraft:modified_jungle"
```
Comment (bageldotjpg) [inline src/generated/resources/data/forge/tags/biomes/wet.json]: Any reason why `modified_jungle_edge` is not here?
Also might want to include `bamboo_jungle` and `bamboo_jungle_hills`

File: src/generated/resources/data/forge/tags/biomes/jungle.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "replace": false,
+  "values": [
+    "minecraft:jungle",
+    "minecraft:jungle_hills",
+    "minecraft:jungle_edge",
+    "minecraft:modified_jungle",
+    "minecraft:modified_jungle_edge"
+  ]
```
Comment (bageldotjpg) [inline src/generated/resources/data/forge/tags/biomes/jungle.json:11]: Missing `bamboo_jungle` and `bamboo_jungle_hills`

File: src/generated/resources/data/forge/tags/biomes/hot.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    "minecraft:deep_warm_ocean",
+    "minecraft:desert_lakes",
+    "minecraft:modified_jungle",
+    "minecraft:modified_jungle_edge",
+    "minecraft:shattered_savanna",
+    "minecraft:shattered_savanna_plateau",
+    "minecraft:eroded_badlands",
+    "minecraft:modified_wooded_badlands_plateau",
+    "minecraft:modified_badlands_plateau"
+  ]
```
Comment (bageldotjpg) [inline src/generated/resources/data/forge/tags/biomes/hot.json:28]: Missing `bamboo_jungle` and `bamboo_jungle_hills`

File: src/generated/resources/data/forge/tags/biomes/hills.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    "minecraft:snowy_taiga_hills",
+    "minecraft:giant_tree_taiga_hills",
+    "minecraft:flower_forest",
+    "minecraft:swamp_hills",
+    "minecraft:ice_spikes",
+    "minecraft:modified_jungle_edge",
+    "minecraft:tall_birch_forest",
+    "minecraft:giant_spruce_taiga_hills",
+    "minecraft:shattered_savanna_plateau",
+    "minecraft:modified_wooded_badlands_plateau"
```
Comment (bageldotjpg) [inline src/generated/resources/data/forge/tags/biomes/hills.json]: Missing `bamboo_jungle_hills`

**Review by bageldotjpg — COMMENTED** [review]
Mostly just seems the Bamboo biomes from 1.14 were missed.

**Comment by GirafiStudios** [comment]
> Mostly just seems the Bamboo biomes from 1.14 were missed.

Thank you very much for your feedback! I did indeed miss the Bamboo biomes. I've added the ones you suggested now.

File: src/generated/resources/data/forge/tags/biomes/rare.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    "minecraft:dark_forest_hills",
+    "minecraft:snowy_taiga_mountains",
+    "minecraft:giant_spruce_taiga",
+    "minecraft:giant_spruce_taiga_hills",
+    "minecraft:modified_gravelly_mountains",
+    "minecraft:shattered_savanna",
+    "minecraft:shattered_savanna_plateau",
+    "minecraft:eroded_badlands",
+    "minecraft:modified_wooded_badlands_plateau",
+    "minecraft:modified_badlands_plateau"
```
Comment (bageldotjpg) [inline src/generated/resources/data/forge/tags/biomes/rare.json]: I don't exactly know how the biomes that go here are determined but `bamboo_jungle` and `bamboo_jungle_hills` may belong here as well

**Review by bageldotjpg — COMMENTED** [review]
Great! Just one thing I missed which was the Bamboo biomes might needing to be in the rare category, but not exactly sure how that category is determined. 

Also, was wondering if maybe there should be a small tag for edge biomes, like there is one hill biomes, containing `jungle_edge`, `modified_jungle_edge` and `mountain_edge`

File: src/generated/resources/data/forge/tags/biomes/rare.json (hunk 6 above)
Comment (GirafiStudios) [reply src/generated/resources/data/forge/tags/biomes/rare.json]: I think it's basically just sub-biomes/edge biomes, so I would agree that the Bamboo biomes would fit under that, yea.

**Comment by GirafiStudios** [comment]
Closing this as there is currently not any plans to pair the Forge Biome registry & the vanilla dynamic Biome registry, as mentioned here https://github.com/MinecraftForge/MinecraftForge/pull/7341#issuecomment-695214020
If things change in the future, feel free to copy any or all of the code from this commit.

**Pull request closed without merging**